packages/apps/job-launcher/server/src/modules/payment/payment.entity.ts

```diff
@@ -7,11 +7,14 @@ import { UserEntity } from '../user/user.entity';
 @Entity({ schema: NS, name: 'payments' })
 @Index(['chainId', 'transaction'], {
   unique: true,
-  where: '(chain_Id IS NOT NULL)',
+  where: '(chain_Id IS NOT NULL AND transaction IS NOT NULL)',
+})
+@Index(['transaction'], {
+  unique: true,
+  where: '(chain_Id IS NULL AND transaction IS NOT NULL)',
 })
-@Index(['transaction'], { unique: true, where: '(chain_Id IS NULL)' })
 export class PaymentEntity extends BaseEntity {
-  @Column({ type: 'varchar', nullable: false })
+  @Column({ type: 'varchar', nullable: true })
```

Comment:
Take care, because some payment records don't have any transaction id, for example the we want to withdraw some amount from user balance and also in this case we can not to leave it null if it has `NOT NULL`.